app/assets/javascripts/directives/task_item.js

```diff
@@ -0,0 +1,55 @@
+angular.module("OpenHq").directive("taskItem", function(Task) {
+  return {
+    restrict: "E",
+    scope: {
+      task: '=',
+      users: '=',
+    },
+    template: JST['templates/directives/task_item'],
+
+    controller: function($scope){
+      $scope.editing = false;
+
+      $scope.startEditing = function($event) {
+        $scope.editing = true;
+        $($event.currentTarget).closest('li').addClass('editing');
+      }
+
+      $scope.stopEditing = function($event) {
+        $scope.editing = false;
+        $($event.currentTarget).closest('li').removeClass('editing');
+      }
+
+      $scope.updateTask = function($event, task) {
+        var $form = $($event.currentTarget),
+            update_params = {
```

Comment:
for the form use `ng-model="task.label"` etc on the form inputs so the form fields get bound to the model then you won’t need to fetch the values